### PR TITLE
Use flyte configuration defaults for sidecar pod spec

### PR DESCRIPTION
**Pull request by katrogan**
# TL;DR
Use flyte configuration defaults for sidecar pod spec.

I tried to be clever and use a patch strategy that doesn't require hand curation for every additional field we write a default for, but I was unsuccessful. I tried dumping pod specs to json and using a merge patch that way but it failed for list fields :( 

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
User reported (thank you @jeevb)

## Tracking Issue
https://github.com/lyft/flyte/issues/548

## Follow-up issue
_NA_

### CODE REVIEW

File: go/tasks/plugins/k8s/sidecar/sidecar.go [hunk 1]
```diff
@@ -27,6 +27,23 @@ const (
 
 type sidecarResourceHandler struct{}
 
+func mergePodSpecs(flyteConfiguredPodSpec, userSpecified *v1.PodSpec) {
```
Comment (kumare3) [inline go/tasks/plugins/k8s/sidecar/sidecar.go:30]: merge seems weird and seems will get out of date everytime we change. What if we change this so that the merging is done by the flytek8s pod_helper.

so we can have a function
flytek8s.UpdatePod(basePod v1.PodSpec)?

Comment (katrogan) [reply go/tasks/plugins/k8s/sidecar/sidecar.go:30]: yeah I agree the manual merge is clunky (see PR comment, I couldn't successfully find a way to dynamically merge the flyte defaults w/ the user specified pod spec). updated to use your suggestion

**Comment by katrogan** [comment]
PTAL @kumare3

**Comment by katrogan** [comment]
@kumare3 I had to remove the `nvidia.com/gpu` resource toleration from the config because an the Interruptible test was complaining - I'm not sure what changed to make that an invalid name

File: copilot/data/download.go [hunk 2]
```diff
@@ -25,7 +25,7 @@ import (
 
 type Downloader struct {
 	format core.DataLoadingConfig_LiteralMapFormat
-	store *storage.DataStore
+	store  *storage.DataStore
```
Comment (kumare3) [inline copilot/data/download.go:28]: Format?

**Review by kumare3 — APPROVED** [review]

File: copilot/data/download.go (hunk 2 above)
Comment (katrogan) [reply copilot/data/download.go:28]: yep, i ran `make goimports` but I can revert though if it's annoying to mess with git blame
